TheArchive.Core/Core/Localization/LocalizationCoreService.cs

```diff
@@ -0,0 +1,140 @@
+                try
+                {
+                    service.SetCurrentLanguage(CurrentLanguage);
+                }
+                catch (Exception ex)
+                {
+                    Logger.Error($"Exception has been thrown in Feature {service.Feature.Name}-->SetCurrentLanguage. {ex}: {ex.Message}");
+                    Logger.Exception(ex);
+                }
+            }
+
+            UpdateAllTexts();
+        }
+
+        public static void UpdateAllTexts()
+        {
+            foreach (KeyValuePair<ILocalizedTextSetter, uint> keyValuePair in m_textSetters)
+            {
+                keyValuePair.Key.SetText(Get(keyValuePair.Value));
+            }
+            foreach (ILocalizedTextUpdater localizedTextUpdater in m_textUpdaters)
+            {
+                localizedTextUpdater.UpdateText();
+            }
+        }
+
+        public static void AddTextSetter(ILocalizedTextSetter textSetter, uint textId)
+        {
+            textSetter.SetText(Get(textId));
+            m_textSetters.Add(textSetter, textId);
+        }
+
+        public static void AddTextSetterDynamic(ILocalizedTextSetter textSetter, uint textId)
+        {
+            textSetter.SetText(Get(textId));
+            m_textSetters.Add(textSetter, textId);
+        }
+
+        public static void SetTextSetter(ILocalizedTextSetter textSetter, uint textId)
+        {
+            textSetter.SetText(Get(textId));
+            m_textSetters[textSetter] = textId;
+        }
+
+        public static void SetTextSetterDynamic(ILocalizedTextSetter textSetter, uint textId)
+        {
+            textSetter.SetText(Get(textId));
+            m_textSetters[textSetter] = textId;
+        }
+
+        public static void AddTextUpdater(ILocalizedTextUpdater textUpdater)
+        {
+            textUpdater.UpdateText();
+            m_textUpdaters.Add(textUpdater);
+        }
+
+        public static string Get(uint id, string defaultValue = "UNKNOWN ID: {0}")
+        {
+            if (!m_texts.TryGetValue(id, out var language) || !language.TryGetValue(CurrentLanguage, out var text) || text.IsNullOrWhiteSpaceOrEmpty())
+            {
+                if (defaultValue == "UNKNOWN ID: {0}")
+                {
+                    defaultValue = string.Format(defaultValue, id);
+                }
+                return defaultValue;
+            }
+            return text;
+        }
+
+        public static string Format(uint id, string defaultValue = "UNKNOWN ID: {0}", params object[] args)
+        {
+            return string.Format(Get(id, defaultValue), args);
+        }
+
+        public static void RegisterLocalizationService(FeatureLocalizationService service)
+        {
+            m_localizationServices.Add(service);
+        }
+
+        public static void Init()
+        {
+            string dir = string.Concat(Path.GetDirectoryName(ArchiveMod.CORE_PATH), $"\\Localization\\{(LoaderWrapper.IsGameIL2CPP() ? "IL2CPP" : "MONO")}\\");
```

Comment:
We probably don't have to separate Mono from IL2CPP as both (should) have the same Features;  
I will have to test that tho
```suggestion
            string dir = string.Concat(Path.GetDirectoryName(ArchiveMod.CORE_PATH), $"\\Localization\\");
```

Reply:
saw your commit 5fc69f44904dab971473b65ea13cb57a978be3aa
fixes the second time this is used, original one in `LocalizationCoreService` is still here haha